File: frontend/src/types/research.ts
export type ResearchStatus = 
  | 'CREATED'
  | 'QUEUED'
  | 'PLANNING'
  | 'SEARCHING'
  | 'COLLECTING_SOURCES'
  | 'ANALYZING'
  | 'VALIDATING'
  | 'GENERATING_REPORT'
  | 'SAVING'
  | 'COMPLETED'
  | 'FAILED'
  | 'CANCELLED';

export interface ResearchJob {
  id: string;
  user_id: string;
  topic: string;
  research_type: string;
  depth: string;
  date_range: string;
  source_preferences?: string[];
  status: ResearchStatus;
  progress: number;
  current_agent?: string;
  current_task?: string;
  error_message?: string;
  started_at?: string;
  completed_at?: string;
  created_at: string;
}

export interface ResearchSource {
  id: string;
  title: string;
  url: string;
  domain: string;
  source_type: string;
  published_date?: string;
  relevance_score: number;
  summary?: string;
  verification_status: 'VERIFIED' | 'UNVERIFIED' | 'REJECTED';
}

export interface ResearchFinding {
  id: string;
  question: string;
  finding: string;
  evidence: string;
  source_ids: string[];
  confidence: number;
  validation_status: 'VALIDATED' | 'UNCERTAIN' | 'DISPUTED';
}

export interface ResearchReport {
  id: string;
  research_job_id: string;
  title: string;
  executive_summary: string;
  content_markdown: string;
  report_type: string;
  status: string;
  version: number;
  saved_to_knowledge_base: boolean;
  sources?: ResearchSource[];
  findings?: ResearchFinding[];
  created_at: string;
  updated_at: string;
}

File: backend/app/schemas/research.py
from typing import Optional, List, Any
from datetime import datetime
from uuid import UUID
from pydantic import BaseModel

class ResearchJobCreate(BaseModel):
    topic: str
    research_type: str = "standard"
    depth: str = "standard"
    date_range: str = "all_time"
    source_preferences: Optional[List[str]] = None

class ResearchJobResponse(BaseModel):
    id: UUID
    user_id: UUID
    topic: str
    research_type: str
    depth: str
    date_range: str
    source_preferences: Optional[List[str]] = None
    status: str
    progress: int
    current_agent: Optional[str] = None
    current_task: Optional[str] = None
    error_message: Optional[str] = None
    started_at: Optional[datetime] = None
    completed_at: Optional[datetime] = None
    created_at: datetime

    class Config:
        from_attributes = True

class ResearchSourceResponse(BaseModel):
    id: UUID
    title: str
    url: str
    domain: str
    source_type: str
    published_date: Optional[str] = None
    relevance_score: float
    summary: Optional[str] = None
    verification_status: str

    class Config:
        from_attributes = True

class ResearchFindingResponse(BaseModel):
    id: UUID
    question: str
    finding: str
    evidence: str
    source_ids: List[Any]
    confidence: float
    validation_status: str

    class Config:
        from_attributes = True
